### PR TITLE
Move VPD jsons to vpd repo

### PR DESCRIPTION
**Pull request by GiridhariKrishnan**
Replacing jsons from ibm_vpd folder to config/ and install json via meson. Setting openbmc to pull from config folder.

Test: Seeing vpd files in system

### CODE REVIEW

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]
Why do we have a PR for this?

**Comment by GiridhariKrishnan** [comment]
Seems like formatting error comments from previous commit, will wait till it get fixed and rebase.

**Review by PriyangaRamasamy — APPROVED** [review]

**Comment by PriyangaRamasamy** [comment]
jenkins run tests please

File: meson.build [hunk 1]
```diff
@@ -192,7 +192,7 @@ install_data(
     install_dir: udev_dir / 'rules.d')
 
 package_datadir = join_paths('share', 'vpd')
-install_subdir('ibm_vpd/fru/', install_dir: package_datadir, strip_directory: true)
+install_subdir('config/ibm/', install_dir: package_datadir, strip_directory: true)
```
Comment (PriyangaRamasamy) [inline meson.build:195]: we need config/vpd_inventory.json to be installed in /usr/share/vpd right?

Comment (GiridhariKrishnan) [reply meson.build:195]: I saw it being there from other path "recipes-phosphor/vpd/ibm-vpd-parser/vpd_inventory.json", but yes I'll add that.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]
These files are moved from GHE path or just moved from ibm_vpd to this new path?
Are all these JSONs latest? and same as what is there in GHE?

**Review by jinuthomas — CHANGES_REQUESTED** [review]
Please move the files from GHE to here , I did not tell you to rename the existing folder. The files used in GHE should be copied to the repo , also I do not see you installing the files in the meson.

**Comment by GiridhariKrishnan** [comment]
> Please move the files from GHE to here , I did not tell you to rename the existing folder. The files used in GHE should be copied to the repo , also I do not see you installing the files in the meson.

The meson file changes installs the jsons:
Here is the log:
./tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/temp/log.do_install.2515620:7:Installing subdir /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/git/config/ibm to /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/image/usr/share/vpd
./tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/temp/log.do_install.2515620:9:Installing /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/git/config/ibm/50001000_v2.json to /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/image/usr/share/vpd
./tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/temp/log.do_install.2515620:10:Installing /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/git/config/ibm/50003000.json to /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/image/usr/share/vpd
./tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/temp/log.do_install.2515620:11:Installing /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/git/config/ibm/50001001.json to /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/image/usr/share/vpd
./tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/temp/log.do_install.2515620:12:Installing /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/git/config/ibm/50001002.json to /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/image/usr/share/vpd
./tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/temp/log.do_install.2515620:13:Installing /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/git/config/ibm/50001001_v2.json to /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/image/usr/share/vpd
./tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/temp/log.do_install.2515620:14:Installing /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/git/config/ibm/50003000_v2.json to /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/image/usr/share/vpd
./tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/temp/log.do_install.2515620:15:Installing /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/git/config/ibm/systems.json to /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/image/usr/share/vpd
./tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/temp/log.do_install.2515620:16:Installing /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/git/config/ibm/50001000.json to /home/girik/NEwOPENBMC/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+5e21fecc31-r1/image/usr/share/vpd

File: meson.build (hunk 1 above)
Comment (PriyangaRamasamy) [reply meson.build:195]: I didn't mean to move vpd_inventory.json into config/ibm folder.
Let the vpd_inventory.json be in config/ directory.
While installing you install config/vpd_inventory.json as well.

File: config/ibm/vpd_inventory.json [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+        "isSystemVpd": true,
+        "extraInterfaces": {
+          "xyz.openbmc_project.Inventory.Item.Chassis": null,
+          "com.ibm.ipzvpd.Location" : {
+              "LocationCode" : "Ufcs"
+          }
+        }
+      }
+    ],
+    "/sys/devices/platform/ahb/ahb:apb/ahb:apb:bus@1e78a000/1e78a480.i2c-bus/i2c-8/8-0051/8-00510/nvmem": [
```
Comment (PriyangaRamasamy) [inline config/ibm/vpd_inventory.json:84]: Please don't pull the default json ( vpd_inventory.json ) from[ GHE](https://github.ibm.com/openbmc/openbmc/blame/2fbed12a41ddda88e66a7a9c45d1ce90262b1205/meta-ibm/recipes-phosphor/vpd/ibm-vpd-parser/vpd_inventory.json#L310 ). looks like something got messed up with this default json as part of rebase in GHE. 
there shouldn't be nvmem paths. 


Please have only these two things in  the default vpd_inventory.json -> 
1. commonInterfaces{ } section completely
2. and under frus{ } section have only motherboard entry ("/sys/bus/i2c/drivers/at24/8-0050/eeprom" )

Comment (GiridhariKrishnan) [reply config/ibm/vpd_inventory.json:84]: Okay. Thanks.

**Comment by GiridhariKrishnan** [comment]
> ci fails due to clang issue https://sys-openbmc-dev-jenkins.swg-devops.com/job/OpenBMC-Dev/job/build/job/github-public/job/repo-ci/job/public-ibm-openbmc-downstream-repo-ci-pr-openpower-vpd-parser/188/consoleFull
> 
> Is this something related to the latest clang 17 issue in 1060 builds?

Probably not, as clang is for C++ related files, these[json] can be due to change in format, probably while installing from meson we have different format.

**Comment by PriyangaRamasamy** [comment]
jenkins run tests please

File: config/ibm/vpd_inventory.json [hunk 3]
```diff
@@ -22,21 +22,24 @@
...
             {
                 "inventoryPath": "/system/chassis/motherboard",
                 "isSystemVpd": true,
-                "systemVpdBackupPath": "/sys/bus/i2c/drivers/at24/7-0050/eeprom",
                 "extraInterfaces": {
                     "xyz.openbmc_project.Inventory.Item.Board.Motherboard": null,
                     "com.ibm.ipzvpd.Location": {
                         "LocationCode": "Ufcs-P0"
-                    },
-                    "xyz.openbmc_project.Inventory.Item": {
```
Comment (PriyangaRamasamy) [inline config/ibm/vpd_inventory.json:38]: why did you remove PrettyName field here?

Comment (GiridhariKrishnan) [reply config/ibm/vpd_inventory.json:38]: I didn't remove it's showing difference for other file, I just had vpd_inventory json edited. I'm not sure why will check that.

Comment (GiridhariKrishnan) [reply config/ibm/vpd_inventory.json:38]: Oh Okay, so the difference here is lesser so, I think git took this. Check with vpd_inventory json will have same things as there.

**Comment by PriyangaRamasamy** [comment]
@GiridhariKrishnan Can you please test if motherboard's sub frus are getting populated on dbus? or can you please share the machine with your image?

File: meson.build (hunk 1 above)
Comment (PriyangaRamasamy) [reply meson.build:195]: As discussed, let the vpd_inventory.json be in config/ibm directory itself. issue resolved

**Review by PriyangaRamasamy — APPROVED** [review]
Looks good to me

File: meson.build [hunk 4]
```diff
@@ -192,7 +192,7 @@ install_data(
     install_dir: udev_dir / 'rules.d')
 
 package_datadir = join_paths('share', 'vpd')
```
Comment (PriyangaRamasamy) [inline meson.build:194]: I think you are missing get_option('datadir'). Please check

**Review by PriyangaRamasamy — CHANGES_REQUESTED** [review]

File: meson.build (hunk 4 above)
Comment (GiridhariKrishna) [reply meson.build:194]: should we need it?
As this is the path I see in logs,
Installing subdir /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/git/config/ibm to /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/image/usr/share/vpd
Installing /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/git/config/ibm/50003000.json to /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/image/usr/share/vpd
Installing /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/git/config/ibm/50003000_v2.json to /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/image/usr/share/vpd
Installing /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/git/config/ibm/vpd_inventory.json to /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/image/usr/share/vpd
Installing /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/git/config/ibm/50001002.json to /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/image/usr/share/vpd
:Installing /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/git/config/ibm/50001001_v2.json to /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/image/usr/share/vpd
Installing /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/git/config/ibm/50001000.json to /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/image/usr/share/vpd
:Installing /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/git/config/ibm/50001001.json to /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/image/usr/share/vpd
1.0+gitAUTOINC+8bd9124c54-r1/temp/log.do_install.2718989:15:Installing /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/git/config/ibm/systems.json to /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/image/usr/share/vpd
1.0+gitAUTOINC+8bd9124c54-r1/temp/log.do_install.2718989:16:Installing /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/git/config/ibm/50001000_v2.json to /home/girik/GHE/openbmc/build/p10bmc/tmp/work/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/ibm-vpd-parser/1.0+gitAUTOINC+8bd9124c54-r1/image/usr/share/vpd

**Comment by PriyangaRamasamy** [comment]
@GiridhariKrishna When I test this changes along with @Alpana07 's change to remove the json installation from recipe, I found that we are missing permissions when installing via meson.

/usr/share/vpd# ls -l
-rw-r--r--    1 root     root        200895 Mar  9  2018 50001000.json
-rw-r--r--    1 root     root        203480 Mar  9  2018 50001000_v2.json
-rw-r--r--    1 root     root        178981 Mar  9  2018 50001001.json
-rw-r--r--    1 root     root        181008 Mar  9  2018 50001001_v2.json
-rw-r--r--    1 root     root        118612 Mar  9  2018 50001002.json
-rw-r--r--    1 root     root        261588 Mar  9  2018 50003000.json
-rw-r--r--    1 root     root        263559 Mar  9  2018 50003000_v2.json
-rwxr-xr-x    1 root     root         47779 Mar  9  2018 50004000.json                // This is installed from recipe file
-rw-r--r--    1 root     root           763 Mar  9  2018 systems.json
-rw-r--r--    1 root     root          2810 Mar  9  2018 vpd_inventory.json

**Comment by PriyangaRamasamy** [comment]
@GiridhariKrishna You need to rebase this PR

File: meson.build [hunk 5]
```diff
@@ -186,7 +186,7 @@ install_data(scripts,
     install_dir: get_option('bindir'))
 
 package_datadir = join_paths('share', 'vpd')
-install_subdir('ibm_vpd/fru/', install_dir: package_datadir, strip_directory: true)
+install_subdir('config/ibm/', install_mode: 'rw-r--r--', install_dir: package_datadir, strip_directory: true)
```
Comment (PriyangaRamasamy) [inline meson.build:189]: Install mode should be -rwxr-xr-x

Comment (GiridhariKrishna) [reply meson.build:189]: Oh Okay, need to check the log, I'll add execute. Thanks.

File: meson.build (hunk 4 above)
Comment (PriyangaRamasamy) [reply meson.build:188]: right.. This package_datadir = join_paths('share', 'vpd') is working to install in the right directory.
Just a question here, Do you know from where it is taking /usr as prefix?

**Review by PriyangaRamasamy — APPROVED** [review]

File: meson.build (hunk 4 above)
Comment (GiridhariKrishnan) [reply meson.build:188]: It's from openbmc recipe, it places to the right path there.

**Review by PriyangaRamasamy — APPROVED** [review]

**Review by SunnySrivastava1984 — APPROVED** [review]

**Review by jinuthomas — APPROVED** [review]
